src/tox/venv.py

```diff
@@ -184,21 +184,21 @@ def _getliveconfig(self):
         develop = self.envconfig.usedevelop
         alwayscopy = self.envconfig.alwayscopy
         deps = []
-        for dep in self._getresolvedeps():
+        for dep in self._get_resolved_dependencies():
             raw_dep = dep.name
             md5 = getdigest(raw_dep)
             deps.append((md5, raw_dep))
         return CreationConfig(md5, python, version, sitepackages, develop, deps, alwayscopy)
 
-    def _getresolvedeps(self):
-        deps = []
-        for dep in self.envconfig.deps:
-            if dep.indexserver is None:
-                res = self.session._resolve_pkg(dep.name)
-                if res != dep.name:
-                    dep = dep.__class__(res)
-            deps.append(dep)
-        return deps
+    def _get_resolved_dependencies(self):
```

Comment:
I realize it's not a public interface, but this breaks [at least this plugin](https://github.com/tox-dev/tox-pip-extensions/blob/3b62f3a331478329e80a16a61610fd72d0d4b47d/tox_pip_extensions.py#L111)

Reply:
Hmm, I was careful to not change anything that's public. If there's a need we should expose it as public, rather than encourage people using a private interface. :thinking:

Reply:
let's not needlessly break this as part of this PR

Reply:
I find it hard to argue against breaking a private function. I'm inclined to break it just to force people not relying on it.

Reply:
I find it hard to advocate for:
- renaming variables and functions unrelated to PR
- needlessly breaking compatibility
- invalidating `git blame`

if you were removing the function or changing its behaviour then fine, break compatibility -- but when there's no replacement, no migration pathway, and in a `tox-dev` plugin I can't see why you'd intentionally advocate for breaking something (that seems borderline malicious!)

Reply:
@asottile 

Both the functions and variables are related to PR: I've refactored the code handling the dependency parsing: As part of the dependency parsing the version parsing is used and as such the version parser is invoked via the ``_getresolvedeps`` path.